### PR TITLE
:wood: log % duration + output_names for FeatureCollection

### LINKED ISSUE #66: improve `get_processor_logs`
Current version:

image

new features:
* add the columns:
   `duration %` -> can be directly calculated from the duration column, so does not need to be stored within the logs itself
   `output_names` -> the output names of the adjusted / newly created series. Can help to improve the function recall?

### PR DESCRIPTION
**Pull request by jvdd**
Together with #73, this should close #66

### CODE REVIEW

**Comment by jvdd** [comment]
At a first glance it looks like logging on Windows returns 0 for very short durations.. (unsure if it returns non-zero for longer durations)

**Comment by jvdd** [comment]
> At a first glance it looks like logging on Windows returns 0 for very short durations.. (unsure if it returns non-zero for longer durations)

https://stackoverflow.com/a/1938096/12180712

Could be due to lower precision of the clock on Windows?
-> perhaps we can overcome this by using
- `time.time_ns()` -> uses more precise integers instead of limited precision floats.
- `time.perf_counter()` -> should use the most accurate clock on the system.

File: tsflex/features/segmenter/strided_rolling.py [hunk 1]
```diff
@@ -494,8 +494,8 @@ def apply_func(self, func: FuncWrapper) -> pd.DataFrame:
         log_window = "manual" if self.window is None else self.window
         logger.info(
```
Comment (jonasvdd) [inline tsflex/features/segmenter/strided_rolling.py:495]: just an idea: is there a flush option (alike with print statements)? That the logging message must be sent before the next code will get hit?

Comment (jvdd) [reply tsflex/features/segmenter/strided_rolling.py:495]: No idea.. not really sure where you aiming at with this? :eyes:

Comment (jonasvdd) [reply tsflex/features/segmenter/strided_rolling.py:495]: That the timing diff is created based on when these logging messages, are outputted; so when they are flushed, we are certain that they are outputted right-away.

Comment (jvdd) [reply tsflex/features/segmenter/strided_rolling.py:495]: Hmm, we use the `elapsed` variable for the timing diff (which is calculated right after the function output is stored in the output dict) - I do not see any problems with this

Comment (jvdd) [reply tsflex/features/segmenter/strided_rolling.py:495]: Flush might however facilitate more direct output towards the end user?

Comment (jvdd) [reply tsflex/features/segmenter/strided_rolling.py:495]: No biggie! :handshake:

**Comment by jvdd** [comment]
> Could be due to lower precision of the clock on Windows? -> perhaps we can overcome this by using
> 
> * `time.time_ns()` -> uses more precise integers instead of limited precision floats - but I think this has still the same underlying precision
> * `time.perf_counter()` -> should use the most accurate clock on the system.

I can now confirm that `time.time_ns()` has the same underlying precision - and thus results in the same error.

Using the `.perf_counter()` seems the way to go!

**Comment by jvdd** [comment]
Ready for review! @jonasvdd

**Review by jonasvdd — APPROVED** [review]
LGTM! very nice work mate 🤝

**Pull request merged**